### PR TITLE
Api 31719 error array 526 v2 part 2

### PR DESCRIPTION
**Pull request by rockwellwindsor-va**
## Summary
Continuing the work from [part 1](https://github.com/department-of-veterans-affairs/vets-api/pull/14859).

* Just converts more errors raises over to the collection
* Updates some code to keep track of disability and secondary disability indexes for the error details.
* Unifies errors to 422 (400 -> 422 for the few 400's there were, this could change)

**The line count will go down when part 1 merges, I branched this off of that since it needed that base to operate at all**

## Related issue(s)
[API-31719](https://jira.devops.va.gov/browse/API-31719)

## Testing done
* RSpec
* Postman

#### Testing notes
Sample testing scenarios for various groups of errors.  Can be grouped together as well.

* Secondary Disabilities
```
        {
          "disabilityActionType": "NEW",
          "name": "Traumatic Brain Injury",
          "classificationCode": "249479",
          "serviceRelevance": "Explanation of service connection",
          "approximateDate": "2019-11-04",
          "ratedDisabilityId": "1234567",
          "diagnosticCode": 9999,
          "isRelatedToToxicExposure": false,
          "secondaryDisabilities": [
            {
              //"name": "Cancer - Musculoskeletal - Elbow",
              //"disabilityActionType": "SECONDARY",
              "classificationCode": "249479",
              //"serviceRelevance": "Injured left knee when brace on right knee failed",
              "approximateDate": "2010-06-11"
            }
          ]
        }
```
* Direct Deposit
```
      "directDeposit": {
        "accountType": "CHECKING",
        //"accountNumber": "123123123123",
        //"routingNumber": "123456789",
        //"financialInstitutionName": "Chase",
        "noAccount": false
      }
```
* Homeless
```
      "homeless": {
        "pointOfContact": "john stewart",
        "pointOfContactNumber": {
          "internationalTelephone": "442-234-5678"
        }
        // "currentlyHomeless": {
        //   "homelessSituationOptions": "FLEEING_CURRENT_RESIDENCE",
        //   "otherDescription": "ABCDEFGHIJKLM"
        // }
      },
```
* Terms of Service
```
          "obligationTermsOfService": {
            "beginDate": "2018-11-24",
            "endDate": "2017-11-17"
          },
```
## Screenshots
Screenshot 2023-12-14 at 11 18 08 AM
Screenshot 2023-12-14 at 11 18 18 AM

## What areas of the site does it impact?
	modified:   modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb
	modified:   modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Comment by FonzMP** [comment]
is the gemfile intended to be in here @rockwellwindsor-va ?

File: modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb [hunk 1]
```diff
@@ -41,11 +41,11 @@ def submit # rubocop:disable Metrics/MethodLength
...
-          if auto_claim.errors.present?
-            raise ::ClaimsApi::Common::Exceptions::Lighthouse::UnprocessableEntity.new(
-              detail: auto_claim.errors.messages.to_s
-            )
-          end
+          # if auto_claim.errors.present?
+          #   raise ::ClaimsApi::Common::Exceptions::Lighthouse::UnprocessableEntity.new(
+          #     detail: auto_claim.errors.messages.to_s
+          #   )
+          # end
```
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb:48]: again, is this intended?

**Review by mchristiansonVA — COMMENTED** [review]
Gemfile.lock got added, needs removal.

File: modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb (hunk 1 above)
Comment (rockwellwindsor-va) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb]: this has been fixed up in the part 1 branch now

**Comment by rockwellwindsor-va** [comment]
Reverted the Gemfile.lock inclusion

**Pull request closed without merging**